### PR TITLE
chore(protocol): rework WorkflowInstanceCreationRecordValue

### LINKED ISSUE #2629: Rework WorkflowInstanceCreationRecordValue
* should extend RecordValueWithVariables interface
 * rename getKey and getInstanceKey to getWorfklowKey and getWorkflowInstanceKey

see https://github.com/zeebe-io/zeebe/issues/2616

### PR DESCRIPTION
**Pull request by miguelpires**
closes #2629

### CODE REVIEW

File: protocol-impl/src/main/java/io/zeebe/protocol/impl/encoding/MsgPackConverter.java [hunk 1]
```diff
@@ -178,4 +178,16 @@ public static String convertJsonSerializableObjectToJson(JsonSerializable record
       throw new RuntimeException(e);
     }
   }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////////
+  ///////////////////////////////////// MAP TO JSON //////////////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////////////////////////
+
+  public static String convertToJson(Map<String, Object> map) {
```
Comment (Zelldon) [inline protocol-impl/src/main/java/io/zeebe/protocol/impl/encoding/MsgPackConverter.java]: Why we need this?

Comment (MiguelPires) [reply protocol-impl/src/main/java/io/zeebe/protocol/impl/encoding/MsgPackConverter.java]: We don't, I forgot to remove that dead code. Fixed it now

**Review by ChrisKujawa — APPROVED** [review]
Thanks :)

**Pull request merged**